src/routes/RemediationDetailsComponents/ExecutionHistoryContent/StatusIcon.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  InProgressIcon,
+  ExclamationCircleIcon,
+  BanIcon,
+  CheckIcon,
+} from '@patternfly/react-icons';
+
+const StatusIcon = ({ status, size = 'sm' }) => {
+  const map = {
+    success: {
+      Icon: CheckIcon,
+    },
+    running: {
+      Icon: InProgressIcon,
+    },
+    failure: {
+      Icon: ExclamationCircleIcon,
+    },
+    canceled: {
+      Icon: BanIcon,
+    },
+  };
```

Comment:
This and the `config` below in `StatusLabel` are very similar and share common data. It might make sense to extract this into the helper.js or maybe a new constants file or something.

Reply:
Theres a bunch more refactoring to do, but I squished them into one file. Post-summit there will be more time to add more improvements because this is used elsewhere. The execution modal is gonna get rewritten completely